File: Data Structures & Algorithms/duplicate-integer/submission-2.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        HashMap dict = new HashMap<>();

        for (int i = 0; i < nums.length; i++){
            if (dict.containsValue(nums[i])){
                return true;
            }
            dict.put(i, nums[i]);
        }

        return false;
    }
}

File: Data Structures & Algorithms/three-integer-sum/submission-2.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        res = []
        for i in range(len(nums)):
            seen = {}
            p = i + 1
            while p < len(nums) - 1:
                seen[nums[p]] = p
                p += 1
                if ((nums[i] + nums[p]) * -1) in seen and sorted([nums[i], nums[p], ((nums[i] + nums[p]) * -1)]) not in res:
                    res.append(sorted([nums[i], nums[p], ((nums[i] + nums[p]) * -1)]))
        return res
